src/main/java/uk/gov/hmcts/reform/ccd/document/am/service/impl/DocumentManagementServiceImpl.java

```diff
@@ -248,31 +282,53 @@ private void prepareRequestForAttachingDocumentToCase(DocumentMetadata documentM
     }
 
     @Override
-    public ResponseEntity<Object> uploadDocuments(List<MultipartFile> files, String classification, List<String> roles,
+    public ResponseEntity<Object> uploadDocuments(List<MultipartFile> files, String classification,
                                                    String caseTypeId, String jurisdictionId) {
+        try {
+            LinkedMultiValueMap<String, Object> bodyMap = new LinkedMultiValueMap<>();
+            HttpHeaders headers = prepareRequestForUpload(classification, caseTypeId, jurisdictionId, bodyMap);
 
-        LinkedMultiValueMap<String, Object> bodyMap = new LinkedMultiValueMap<>();
-        HttpHeaders headers = prepareRequestForUpload(classification, roles, caseTypeId, jurisdictionId,bodyMap);
-
-        for (MultipartFile file : files) {
-            if (!file.isEmpty()) {
-                bodyMap.add(FILES, file.getResource());
+            for (MultipartFile file : files) {
+                if (!file.isEmpty()) {
+                    bodyMap.add(FILES, file.getResource());
+                }
             }
-        }
-        String docUrl = String.format("%s/documents", documentURL);
+            String docUrl = String.format("%s/documents", documentURL);
 
-        HttpEntity<LinkedMultiValueMap<String, Object>> requestEntity = new HttpEntity<>(bodyMap, headers);
-        ResponseEntity<Object> uploadedDocumentResponse = restTemplate
-            .postForEntity(docUrl, requestEntity, Object.class);
+            HttpEntity<LinkedMultiValueMap<String, Object>> requestEntity = new HttpEntity<>(bodyMap, headers);
 
-        if (HttpStatus.OK.equals(uploadedDocumentResponse.getStatusCode()) && null != uploadedDocumentResponse
-            .getBody()) {
-            formatUploadDocumentResponse(caseTypeId, jurisdictionId, uploadedDocumentResponse);
-        }
+            ResponseEntity<Object> uploadedDocumentResponse = restTemplate
+                .postForEntity(docUrl, requestEntity, Object.class);
 
-        return ResponseEntity
-            .status(uploadedDocumentResponse.getStatusCode())
-            .body(uploadedDocumentResponse.getBody());
+            if (HttpStatus.OK.equals(uploadedDocumentResponse.getStatusCode()) && null != uploadedDocumentResponse
+                .getBody()) {
+                formatUploadDocumentResponse(caseTypeId, jurisdictionId, uploadedDocumentResponse);
+            }
+
+            return ResponseEntity
+                .status(uploadedDocumentResponse.getStatusCode())
+                .body(uploadedDocumentResponse.getBody());
+        } catch (HttpClientErrorException exception) {
+            if (HttpStatus.NOT_FOUND.equals(exception.getStatusCode())) {
```

Comment:
Create common method to check all exceptions inside catch block.As, it is common check and using in every api implementation.It will reduce lots of duplicate code and shrink the file size and length.

Reply:
Yes, It is very much required. It is also captured in sonar duplication.

Reply:
It is implemented.